src/null-safety/migrating.md

```diff
@@ -0,0 +1,397 @@
+
+1. Modify the SDK constraints in the package's `pubspec.yaml` file,
+   setting the language version to 2.12 or later:
+   ```yaml
+environment:
+  sdk: >=2.12.0 <3.0.0
+```
+
+2. Run `dart pub get`. <br>
+   ```terminal
+$ dart pub get
+```
+
+   Running `dart pub get` with a lower constraint of 2.12.0
+   sets the default language version of
+   every library in your package to 2.12,
+   opting them all in to null safety.
+
+
+3. Open the package in your IDE. <br>
+   You're likely to see a lot of analysis errors.
+   That's OK.
+
+4. _Optional:_ Add a [language version comment][] to the top of
+   any Dart files that you don't want to consider during your current migration:
+   ```dart
+// @dart=2.9
+```
+
+   Using language version 2.9 for a library that's in a 2.12 package
+   enables mixed mode and can reduce analysis errors (red squiggles).
+   However, **mixed mode reduces the
+   information the analyzer can use.**
+   For example, the analyzer might assume a
+   parameter type is non-nullable,
+   even though a 2.9 file might pass in a null value.
+
+5. Migrate the code of each Dart file,
+   using the analyzer to identify static errors. <br>
+   Add `?`, `!`, `required`, `late` and refactoring as needed
+   to improve your code and eliminate static errors.
+
+[language version comment]: /guides/language/evolution#per-library-language-version-selection
+
+**_[PENDING: Do we need to talk about `pub get` generating a [package config][]?
+That link isn't as helpful as it could be.
+Should/do we cover this on dart.dev?]_**
+
+[package config]: https://pub.dev/packages/package_config
+
+
+## Test your code
+
+If you already have tests, run them.
+You might need to update tests that expect null values.
+
+**_[PENDING:
+See https://github.com/flutter/flutter/wiki/Experimenting-with-null-safety-in-Flutter
+for details (on testing and otherwise; it's sort of this doc, but for pre-beta.
+]_**
+
+
+## Publish your code
+
+We encourage you to publish non-app packages as soon as you migrate.
+
+
+### Publishing a migrated package
+
+If your package is already on pub.dev, publish it.
+Treat the migrated version of your package as a breaking change:
+
+* If your package is already at 1.0.0 or greater,
+  increase the major version. <br>
+  For example, if the previous version is 2.3.2,
+  the new version is 3.0.0.
+
+* If your package hasn't reached 1.0.0 yet,
+  increase the minor version or update the version to 1.0.0. <br>
+  For example, if the previous version is 0.3.2,
+  the new version is 0.4.0 or 1.0.0.
+
+
+### Deploying a migrated app
+
+Although you can deploy your app at any time,
+we generally recommend waiting for a stable SDK release that
+supports null safety without flags.
+
+
+## Welcome to null safety
+
+If you made it this far,
+you should have a fully migrated, null-safe Dart package.
+If all of the packages you depend on are migrated too,
+then your program is fully sound with respect to null-reference errors.
```

Comment:
I'm not sure we should use "fully sound". There is no partly sound. You can be either sound or unsound.